dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/TasksRunner.java

```diff
@@ -70,18 +72,28 @@ public TasksRunner(
   }
 
   private TaskRunContext createContext(
-      OutputHandleFactory sinkFactory, Handle handle, int threadPoolSize, Impl state) {
-    return new TaskRunContext(sinkFactory, handle, threadPoolSize) {
-      @Override
-      public TaskState getTaskState(Task<?> task) {
-        return state.getTaskState(task);
-      }
-
-      @Override
-      public <T> T runChildTask(Task<T> task) throws MetadataDumperUsageException {
-        return handleTask(task);
-      }
-    };
+      OutputHandleFactory sinkFactory,
+      Handle handle,
+      int threadPoolSize,
+      Impl state,
+      ConnectorArguments arguments) {
+    return new TaskRunContext(
+        sinkFactory,
+        handle,
+        threadPoolSize,
+        new TaskRunContextOps() {
```

Comment:
Optional: IMO it would improve clarity to move this into a helper method or an inner class.

Reply:
I've made `TasksRunner` implement `TaskRunContextOps`. This way I can simply pass `this` here.